docusaurus/docs/tutorial/02.md

```diff
@@ -1,125 +1,386 @@
+```javascript
+/* src/redux/profile-page/initial-state.js */
+
+const initialState = {
+    /* Some other states */,
+    name: "User"
+};
+
+export default initialState;
 ```
-import React, { Component } from 'react';
-import AsyncComponent from "@treats/component/async-component";
-import './todo.css';
-import List from './list';
 
-class Todo extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      term: '',
-      items: []
-    };
-  }
-
-  render() {
-    return (
-      <div>
-        <form className="App" onSubmit={this.onSubmit}>
-          <input value={this.state.term} onChange={this.onChange} />
-          <button>Submit</button>
-        </form>
-        <List items={this.state.items} />
-      </div>
-    );
-  }
-}
-export default AsyncComponent(module, Todo);
+7. Alright, all set in our Redux part. Let's continue into the next section.
+
+## Tweaking The Filesystem Hooks
+Treats provide [filesystem hooks][filesystem-hooks] for easy custom configuration and Redux is one of it. We already create our reducers for profile page, but the whole app only need a single reducer, since Treats only have a single store. Fortunately, there's some helpful method called `combineReducers` and Treats already wrap it in its dependencies.
+
+1. Create folder `_redux` under `src` and create `index.js` inside it.
+```text
+src
+|-- ...
+|-- _redux
+    |-- reducer.js
 ```
 
-3. Hold a minute! Don't we need a handler function for `onSubmit` and `onChange`? Good question, now lets implement our handler function
+2. Write your `reducer.js` file
+```javascript
+/* src/_redux/reducer.js */
+
+import { combineReducers } from "@treats/redux";
+/* Other reducer */
+import ProfileReducer from "../redux/profile-page";
+
+export default combineReducers({
+    /* Other reducer */
+    profile: ProfileReducer
+});
 ```
-class Todo extends Component {
-  ....
-  onChange = (event) => {
-    this.setState({ term: event.target.value });
-  }
-
-  onSubmit = (event) => {
-    event.preventDefault();
-    this.setState({
-      term: '',
-      items: [...this.state.items, this.state.term]
-    });
-  }
-  ....
+
+3. All set for filesystem hooks, now let's get back to our unfinished components.
+
+## `mapStateToProps` and `mapDispatchToProps`
+Okay in the first section, we promise to implement our `mapStateToProps`. Here we are:
+
+1. Implement our `mapStateToProps` in `profile.js`
+```javascript
+/* src/page/profile-page/profile.js */
+
+...
+
+const mapStateToProps = state => ({
+    name: state.profile.name
+});
+
+```
+
+2. Realize that our submit button doesn't do anything when clicked? Since we will push our new name to Redux, we are gonna dispatch `updateName` we create earlier. Here, we will need to implement `mapDispatchToProps`.
+```javascript
+/* src/page/profile-page/profile.js */
+
+/* Other import */
+import { profileActions } from "../../redux/profile-page";
```

Comment:
This will be better using `@redux` and introducing alias.js in `treats.config.js`